expression/constraint_propagation.go

```diff
@@ -250,7 +250,7 @@ func ruleColumnOPConst(ctx sessionctx.Context, i, j int, exprs *exprSet) {
 			return
 		}
 	}
-	if !col1.Equal(ctx, col2) {
+	if col1.ColName.L != col2.ColName.L {
```

Comment:
`col1.Equal(ctx, col2)` use `UniqueID` for comparison, `UniqueID` is generated and always not same.
expr1 = Parse("c < xx")
expr2 = Parse("to_days(c) > yy")
`expr1.c` Equal `expr2.c` returns false, even the `c` is the same column

Reply:
@winoros Do you have any suggestion?

Reply:
Why convert `to_days(c)` from AST to tidb internal expression will cause the column's unique id different?
Is it because you call SimpleExpressionRewriter? Where did you call it?

Reply:
Because convert `to_days(c) > yy` and "c < xx" are not in one expression.
They are obtained from two different expressions @winoros